aten/src/ATen/core/Generator.h

```diff
@@ -74,6 +75,8 @@ struct CAFFE2_API Generator {
   // See Note [Acquire lock when using random generators]
   std::mutex mutex_;
 
+  virtual DispatchKeySet key_set() const = 0;
```

Comment:
We should just store the DispatchKeySet inline in Generator and devirtualize this function. Ordinarily I wouldn't die on this hill, but I've had a bad experience with virtual methods on TensorImpl being impossible to remove post facto, so I'd prefer if we did it the less flexible way to start here.